### PR TITLE
YubiKey Attestation

### PR DESCRIPTION
**Pull request by maraino**
### Description

This PR adds support for attestation using YubiKeys. It also adds the default template used with ACME device-attest-01.

### CODE REVIEW

File: kms/yubikey/yubikey.go [hunk 1]
```diff
@@ -190,6 +218,35 @@ func (k *YubiKey) CreateSigner(req *apiv1.CreateSignerRequest) (crypto.Signer, e
...
+	if err != nil {
+		return nil, err
+	}
+
+	cert, err := k.yk.Attest(slot)
+	if err != nil {
+		return nil, errors.Wrap(err, "error attesting slot")
+	}
+
+	intermediate, err := k.yk.Certificate(slotAttestation)
```
Comment (hslatman) [inline kms/yubikey/yubikey.go:238]: I guess this does the same as `k.yk.AttestationCertificate()`?

File: kms/yubikey/yubikey.go [hunk 2]
```diff
@@ -190,6 +218,35 @@ func (k *YubiKey) CreateSigner(req *apiv1.CreateSignerRequest) (crypto.Signer, e
...
+//
+// Notice: This API is EXPERIMENTAL and may be changed or removed in a later
+// release.
+func (k *YubiKey) CreateAttestation(req *apiv1.CreateAttestationRequest) (*apiv1.CreateAttestationResponse, error) {
+	slot, err := getSlot(req.Name)
+	if err != nil {
+		return nil, err
+	}
+
+	cert, err := k.yk.Attest(slot)
```
Comment (hslatman) [inline kms/yubikey/yubikey.go:233]: Callers of this function would call this after `k.yk.GenerateKey` with the name of the slot in which the key was created, correct?

**Review by hslatman — APPROVED** [review]

File: kms/yubikey/yubikey.go (hunk 1 above)
Comment (maraino) [reply kms/yubikey/yubikey.go:238]: Yes, I didn't realize that method was present.
```go
// AttestationCertificate returns the YubiKey's attestation certificate, which
// is unique to the key and signed by Yubico.
func (yk *YubiKey) AttestationCertificate() (*x509.Certificate, error) {
	return yk.Certificate(slotAttestation)
}
```

File: kms/yubikey/yubikey.go (hunk 2 above)
Comment (maraino) [reply kms/yubikey/yubikey.go:233]: For example, you can create a key using:

```console
$ step kms create yubikey:slot-id=85
-----BEGIN PUBLIC KEY-----
...
-----END PUBLIC KEY-----
$ step kms attest yubikey:slot-id=85
-----BEGIN CERTIFICATE-----
...
-----END CERTIFICATE-----
-----BEGIN CERTIFICATE-----
...
-----END CERTIFICATE-----
```

**Pull request merged**